sequana/itol.py

```diff
@@ -62,49 +77,68 @@ def __init__(self, tree):
             "display_mode": 2,  # rotation
             "ignore_branch_length": 1,
             "line_width": 5,
-            #"datasets_visible": "0,1",
-            'bootstrap_display': 1,
-            'bootstrap_type': 2,
-            'bootstrap_label_size': 32
+            # "datasets_visible": "0,1",
+            "bootstrap_display": 1,
+            "bootstrap_type": 2,
+            "bootstrap_label_size": 32,
         }
 
+        if APIkey:
+            self.params["APIkey"] = APIkey
+            self.params["projectName"] = projectName
+
     def add_file(self, filename):
         self.itol.add_file(filename)
-        N = len(self.itol.files) - 1   # remove the input tree file
-        self.params['datasets_visible'] = ",".join([str(x) for x in range(0, N)])
+        N = len(self.itol.files) - 1  # remove the input tree file
+        self.params["datasets_visible"] = ",".join([str(x) for x in range(0, N)])
 
     def upload(self):
+        self.itol.params = self.params
         self.status = self.itol.upload()
-        assert self.status, "Something is wrong with your input tree"
+        if self.status is False:
```

Comment:
`if not self.status` 😛

Reply:
Je sais pas si le upload peut raise une error si t'as un soucis de connection ?

Reply:
A priori t'es pas a l'abri d'avoir des erreur de `requests.post` en utilisant le `itol.upload`